client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerParamsOverrideTest.java

```diff
@@ -0,0 +1,117 @@
+package org.infinispan.client.hotrod.event;
+
+import io.netty.buffer.ByteBuf;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.annotation.ClientListener;
+import org.infinispan.client.hotrod.exceptions.HotRodClientException;
+import org.infinispan.client.hotrod.impl.RemoteCacheImpl;
+import org.infinispan.client.hotrod.impl.operations.OperationsFactory;
+import org.infinispan.client.hotrod.impl.protocol.Codec28;
+import org.infinispan.client.hotrod.test.SingleHotRodServerTest;
+import org.testng.annotations.Test;
+
+import java.lang.annotation.Annotation;
+import java.util.Deque;
+import java.util.LinkedList;
+
+import static org.infinispan.test.TestingUtil.replaceField;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertTrue;
+
+@Test(groups = "functional", testName = "client.hotrod.event.ClientListenerParamsOverrideTest")
+public class ClientListenerParamsOverrideTest extends SingleHotRodServerTest {
+
+   private TestCodec testCodec;
+
+   @Override
+   protected RemoteCacheManager getRemoteCacheManager() {
+      final RemoteCacheManager remote = super.getRemoteCacheManager();
+      final RemoteCacheImpl<?, ?> cache = (RemoteCacheImpl<?, ?>) remote.getCache();
+      final OperationsFactory opsFactory = cache.getOperationsFactory();
+      testCodec = new TestCodec();
+      replaceField(testCodec, "codec", opsFactory, OperationsFactory.class);
+      return remote;
+   }
+
+   public void testClientListenerAnnotation() {
+      final RemoteCache<?, ?> cache = remoteCacheManager.getCache();
+      try {
+         cache.addClientListener(new AnnotatedListener());
+      } catch (HotRodClientException e) {
+         // Expected since there are no factories
+         final ClientListener listener = testCodec.clientListeners.pop();
+         assertEquals("a", listener.filterFactoryName());
+         assertEquals("b", listener.converterFactoryName());
+         assertTrue(listener.useRawData());
+         assertTrue(listener.includeCurrentState());
+      }
+   }
+
+   public void testClientListenerParameter() {
+      final RemoteCache<?, ?> cache = remoteCacheManager.getCache();
+      try {
+         ClientListener listenerParams = new ClientListener() {
```

Comment:
This does not implement hashcode and equals as per java.lang.annotation.Annotation contract. Especially hashcode.
Sure that does not harm us but it just leaves me the impression we are stretching this annotation thing more than we should.